app/models/rent.rb

```diff
@@ -1,5 +1,22 @@
 class Rent < ApplicationRecord
   validates :init_date, :end_date, presence: true
+  validate :valid_init_end_dates, :overlap_dates
   belongs_to :user, optional: true
   belongs_to :book, optional: true
+
+  scope :between_from_dates, -> { where('DATE(?) BETWEEN init_date AND end_date', Time.current) }
+
+  def valid_init_end_dates
+    init_date < end_date
+  end
+
+  def overlap_dates
+    Rent.where(book_id: id).where.not(book_id: id).where(
```

Comment:
`where(book_id: id).where.not(book_id: id)` ????
i think you mean
`where(book_id: book_id).where.not(id: id)`